src/freenet/support/api/BooleanCallback.java

```diff
@@ -3,11 +3,21 @@
  * http://www.gnu.org/ for further details of the GPL. */
 package freenet.support.api;
 
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
 import freenet.config.ConfigCallback;
 
 /**
  * A callback to be called when a config value of integer type changes.
  * Also reports the current value.
  */
 public abstract class BooleanCallback extends ConfigCallback<Boolean> {
+  public static BooleanCallback fromBoolean(Supplier<Boolean> get, Function<Boolean, Exception> set) {
+    return (BooleanCallback) ConfigCallback.from(get, set);
```

Comment:
Uhm… this is not how casting works. `ConfigCallback.from` returns a `ConfigCallback` object, no amount of casting will magically turn it into a `BooleanCallback`.

You know, you would have caught that if you had written tests for this. 😠